--- a/finalforeach/cosmicreach/world/ZoneLoaders.java
+++ b/finalforeach/cosmicreach/world/ZoneLoaders.java
@@ -10,8 +_,9 @@
 import finalforeach.cosmicreach.util.logging.Logger;
 
 public class ZoneLoaders {
+	private static final org.slf4j.Logger LOGGER = org.slf4j.LoggerFactory.getLogger("Cosmic Reach | World loaders");
 	public static final ZoneLoaders INSTANCE = new ZoneLoaders();
-	private PauseableThread worldGenThread;
+	public PauseableThread worldGenThread;
 	private ObjectMap<Zone, IZoneLoader> zoneLoaders = new ObjectMap<>();
 
 	public void continueWorldGenThread() {
@@ -23,7 +_,7 @@
 			IZoneLoader izoneloader = GameSingletons.zoneLoaderCreator.apply(zone);
 			this.zoneLoaders.put(zone, izoneloader);
 			this.continueWorldGenThread();
-			Logger.info("Added zone loader: " + izoneloader.getClass().getSimpleName() + " for zone: " + zone.zoneId);
+			LOGGER.info("Added zone loader: " + izoneloader.getClass().getSimpleName() + " for zone: " + zone.zoneId);
 		}
 	}
 
