--- a/finalforeach/cosmicreach/networking/packets/blocks/PlaceBlockPacket.java
+++ b/finalforeach/cosmicreach/networking/packets/blocks/PlaceBlockPacket.java
@@ -46,7 +_,7 @@
 
 	@Override
 	public void handle(NetworkIdentity identity, ChannelHandlerContext ctx) {
-		if (!identity.isClient()) {
+		if (!identity.isClient() && com.github.puzzle.paradox.game.server.ParadoxServerSettings.canPlaceBlock) {
 			Zone zone = identity.getZone();
 			Player player = identity.getPlayer();
 			if (player != null && zone != null) {
@@ -65,6 +_,8 @@
 								itemslot.setDirty(true);
 								player.inventory.syncDirtySlots(0, identity);
 							} else {
+								if(!com.github.puzzle.paradox.game.player.PlayerChecks.blockChecks(identity,blockPos))
+									return;
 								BlockEventTrigger[] ablockeventtrigger = this.targetBlockState.getTrigger("onPlace");
 								if (ablockeventtrigger != null) {
 									BlockEventArgs blockeventargs = new BlockEventArgs();
